src/java.base/share/classes/java/net/Inet6Address.java

```diff
@@ -477,6 +483,146 @@ public static Inet6Address getByAddress(String host, byte[] addr,
         throw new UnknownHostException("addr is of illegal length");
     }
 
+    /**
+     * Creates an {@code InetAddress} based on the provided textual representation of
+     * an IPv6 address.
+     * <p>The following IPv6 address {@linkplain Inet6Address##format
+     * textual representations} are supported by this method:
+     * {@snippet :
+     *  // The full IPv6 form
+     *  Inet6Address.ofLiteral("1080:0:0:0:8:800:200C:417A") ==> /1080:0:0:0:8:800:200c:417a
+     *
+     *  // The compressed IPv6 form with multiple groups of 16-bits of
+     *  // zero replaced with "::"
+     *  Inet6Address.ofLiteral("1080::8:800:200C:417A") ==> /1080:0:0:0:8:800:200c:417a
+     *
+     *  // IPv4-mapped IPv6 form
+     *  Inet6Address.ofLiteral("::FFFF:129.144.52.38") ==> /129.144.52.38
+     *
+     *  // IPv4-compatible IPv6 form
+     *  Inet6Address.ofLiteral("::129.144.52.38") ==> /0:0:0:0:0:0:8190:3426
+     *
+     *  // IPv6 scoped address form with scope-id as numeric identifier
+     *  Inet6Address.ofLiteral("fe80::1%1") ==> /fe80:0:0:0:0:0:0:1%1
+     *
+     *  // IPv6 scoped address with scope-id as string
+     *  Inet6Address.ofLiteral("fe80::1%en0") ==> /fe80:0:0:0:0:0:0:1%en0
+     * }
+     * All IPv6 address literal forms listed above are also supported when enclosed in
+     * square brackets.
+     * <p>If the provided address literal cannot represent a valid IP address an
+     * {@code IllegalArgumentException} is thrown. For instance, if an IPv6 scoped
+     *  address literal contains a scope-id that doesn't map to any network interface
+     *  on the system, or if a scope-id is present in an IPv4-mapped IPv6 address literal.
+     * <p>This method doesn't block, i.e. no reverse lookup is performed.
+     *
+     * @param ipv6AddressLiteral the textual representation of an IPv6 address.
+     * @return an {@link Inet6Address} object with no hostname set, and constructed
+     *         from the IPv6 address literal.
+     * @throws IllegalArgumentException if the {@code ipv6AddressLiteral} cannot be
+     *         parsed as an IPv6 address literal.
+     * @throws NullPointerException if the {@code ipv6AddressLiteral} is {@code null}.
+     */
```

Comment:
Similar comment to Inet4Address. The snippet should be incorporated into the general class docs section "Textual representation of IP addresses" and the apidoc here should limit itself to any limitations that only apply to this method imo.

Reply:
That is a good point Michael. The snippets were moved to class-level javadocs, and were modified a bit to limit the number of examples, and other small changes. Pushed as [e95858e](https://github.com/openjdk/jdk/pull/15775/commits/e95858ebb6039e9b5bbbca2a4730fcbed78747ca).